apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -115,6 +117,16 @@ export class PublishAction extends BaseRushAction {
       `the package is older than the current latest, so in publishing workflows for older releases, providing ` +
       `a tag is important. When hotfix changes are made, this parameter defaults to 'hotfix'.`
     });
+    this._npmAccessLevel = this.defineChoiceParameter({
+      alternatives: ['public', 'restricted'],
+      parameterLongName: '--set-access-level',
+      parameterShortName: undefined,
+      description:
+      `The access option to pass to npm publish. By default NPM will publish scoped packages with an access ` +
```

Comment:
Technically, when publishing a package, by default NPM preserves whatever access level was specified previously.  So it might be more clear to say that for a scoped package, this setting can be used to change the access level during publishing (instead of having to separately run `npm access public my-package`).  Is that correct?

Reply:
Makes sense.  However we also have a closely related problem for the internal SharePoint repo:  For procedural reasons, packages under the `@microsoft` scope are initially created in a `restricted` state (by publishing an empty tarball), and then they are promoted to `public` when the first real release is published.  So your `--set-access-level public` would not work in that case.

Maybe we could support a `--set-access-level force-public` to handle that case?  

Thus `rush public --set-access-level public`  would pass `--set-access-level` to NPM, whereas `rush public --set-access-level force-public`  would instead invoke `npm access` after `npm publish`.

Reply:
cc: @ecraig12345

Reply:
> > this setting can be used to change the access level during publishing (instead of having to separately run npm access public my-package).
> 
> My understanding from the [NPM documentation](https://docs.npmjs.com/cli/access.html#details) is that this flag can _only_ be used to change a scoped package's access level during _initial_ publish. Otherwise, you're advised to use the `npm access` command.

Revisited this... It turns out that my understanding of NPM's documentation was incorrect, mainly this line:

>after the initial publish.

I've confirmed locally that the `--access` flag allows you to toggle the scoped package's access level per publish (`restricted` to `public` and vice versa) _any number of times_.

NPM's documentation means that in place of the flag, you can run `npm access` _after_ the publish step. See the below test output where I'm using `npm publish --access` for multiple publishes of the same package `@kevintcoughlin/test-package-1`:

```shell
~\test-package-1> npm search @kevintcoughlin/test-package-1
NAME                      | DESCRIPTION          | AUTHOR          | DATE       | VERSION  | KEYWORDS
@kevintcoughlin/test-pack |                      | =kevintcoughlin | 2019-02-11 | 1.0.3    |
age-1                     |                      |                 |            |          |

~\test-package-1> npm version patch
v1.0.4

~\test-package-1> npm publish --access restricted
+ @kevintcoughlin/test-package-1@1.0.4

~\test-package-1> npm search @kevintcoughlin/test-package-1
NAME                      | DESCRIPTION          | AUTHOR          | DATE       | VERSION  | KEYWORDS
@kevintcoughlin/test-pack |                      | =kevintcoughlin | 2019-02-11 | 1.0.3    |
age-1

~\test_publish\test-package-1> npm version patch
v1.0.5

~\test_publish\test-package-1> npm publish --access public
+ @kevintcoughlin/test-package-1@1.0.5

~\test_publish\test-package-1> npm search @kevintcoughlin/test-package-1
NAME                      | DESCRIPTION          | AUTHOR          | DATE       | VERSION  | KEYWORDS
@kevintcoughlin/test-pack |                      | =kevintcoughlin | 2019-02-11 | 1.0.5    |
age-1
```

**Version & Access Level**

- `1.0.3`: `public`
- `1.0.4`: `restricted`
- `1.0.5`: `public`

Reply:
@octogonz, I'm going to resolve https://github.com/Microsoft/web-build-tools/pull/1019#discussion_r247287572 since it seems unnecessary given the investigation above. Please re-open if you feel otherwise.